client/am/console/src/main/java/org/apache/syncope/client/console/commons/AMRealmPolicyProvider.java

```diff
@@ -77,7 +77,7 @@ public void add(final RealmTO realmTO, final RepeatingView view) {
                 new PropertyModel<>(realmTO, "authPolicy"),
                 false);
         authPolicy.setChoiceRenderer(new PolicyRenderer(authPolicies));
-        authPolicy.setChoices(authPolicies.keySet().stream().collect(Collectors.toList()));
+        authPolicy.setChoices(new ArrayList<>(authPolicies.keySet()));
```

Comment:
Can you provide some reference to explain that in general

```java
new ArrayList<>(collection);
```

shall be preferred over

```java
collection.stream().collect(Collectors.toList())
```

?

Thanks.

Reply:
Sure. The following comes to mind:

`new ArrayList<>(collection)` directly copies elements from the given collection using the `addAll()` method. This is an optimized operation. The stream variant of course has the overhead of creating a stream, iterating over elements, and creating a collector. The proposed syntax is also more compact with fewer calls, and you can also immediately see the list implementation used, unlike the toList() call where the implementation is somewhat unclear unless you look up docs or are intimately familiar with what is returned. 

Streaming would be useful if the intention is to do extra filtering or other intermediate ops, or if parallel processing is required, etc.

Reply:
Thank you, it makes sense.